File: src/main/java/com/connectflow/model/TransactionEditHistory.java
package com.connectflow.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "transaction_edit_history")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TransactionEditHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @JdbcTypeCode(SqlTypes.CHAR)
    @Column(columnDefinition = "CHAR(36)")
    private UUID id;

    @JdbcTypeCode(SqlTypes.CHAR)
    @Column(name = "transaction_id", nullable = false, columnDefinition = "CHAR(36)")
    private UUID transactionId;

    @Column(name = "pawn_id")
    private String pawnId;

    @JdbcTypeCode(SqlTypes.CHAR)
    @Column(name = "edited_by", nullable = false, columnDefinition = "CHAR(36)")
    private UUID editedBy;

    @Column(name = "edited_by_name")
    private String editedByName;

    @Column(name = "edit_type", nullable = false)
    private String editType; // "STATUS", "DETAILS", "BLOCK", "REMARKS"

    // Previous values
    @Column(name = "previous_status")
    private String previousStatus;

    @Column(name = "previous_address", length = 500)
    private String previousAddress;

    @Column(name = "previous_phone")
    private String previousPhone;

    @Column(name = "previous_loan_amount", precision = 12, scale = 2)
    private BigDecimal previousLoanAmount;

    @JdbcTypeCode(SqlTypes.CHAR)
    @Column(name = "previous_interest_rate_id", columnDefinition = "CHAR(36)")
    private UUID previousInterestRateId;

    @Column(name = "previous_period_months")
    private Integer previousPeriodMonths;

    @Column(name = "previous_maturity_date")
    private LocalDate previousMaturityDate;

    @Column(name = "previous_remarks", length = 1000)
    private String previousRemarks;

    // New values
    @Column(name = "new_status")
    private String newStatus;

    @Column(name = "new_address", length = 500)
    private String newAddress;

    @Column(name = "new_phone")
    private String newPhone;

    @Column(name = "new_loan_amount", precision = 12, scale = 2)
    private BigDecimal newLoanAmount;

    @JdbcTypeCode(SqlTypes.CHAR)
    @Column(name = "new_interest_rate_id", columnDefinition = "CHAR(36)")
    private UUID newInterestRateId;

    @Column(name = "new_period_months")
    private Integer newPeriodMonths;

    @Column(name = "new_maturity_date")
    private LocalDate newMaturityDate;

    @Column(name = "new_remarks", length = 1000)
    private String newRemarks;

    @Column(name = "block_reason", length = 1000)
    private String blockReason;

    @Column(name = "police_report_number")
    private String policeReportNumber;

    @Column(name = "police_report_date")
    private LocalDate policeReportDate;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: next/backend/src/model/TransactionEditHistory.js
'use strict';
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const TransactionEditHistory = sequelize.define('TransactionEditHistory', {
  id:                    { type: DataTypes.CHAR(36),        primaryKey: true, defaultValue: DataTypes.UUIDV4 },
  transactionId:         { type: DataTypes.CHAR(36),        field: 'transaction_id' },
  pawnId:                { type: DataTypes.STRING,          field: 'pawn_id' },
  editedBy:              { type: DataTypes.CHAR(36),        field: 'edited_by' },
  editedByName:          { type: DataTypes.STRING,          field: 'edited_by_name' },
  editType:              { type: DataTypes.STRING(50),      field: 'edit_type' },

  previousStatus:        { type: DataTypes.STRING(50),      field: 'previous_status' },
  newStatus:             { type: DataTypes.STRING(50),      field: 'new_status' },

  previousAddress:       { type: DataTypes.STRING,          field: 'previous_address' },
  newAddress:            { type: DataTypes.STRING,          field: 'new_address' },

  previousPhone:         { type: DataTypes.STRING,          field: 'previous_phone' },
  newPhone:              { type: DataTypes.STRING,          field: 'new_phone' },

  previousLoanAmount:    { type: DataTypes.DECIMAL(18, 2),  field: 'previous_loan_amount' },
  newLoanAmount:         { type: DataTypes.DECIMAL(18, 2),  field: 'new_loan_amount' },

  previousInterestRateId:{ type: DataTypes.CHAR(36),        field: 'previous_interest_rate_id' },
  newInterestRateId:     { type: DataTypes.CHAR(36),        field: 'new_interest_rate_id' },

  previousPeriodMonths:  { type: DataTypes.INTEGER,         field: 'previous_period_months' },
  newPeriodMonths:       { type: DataTypes.INTEGER,         field: 'new_period_months' },

  previousMaturityDate:  { type: DataTypes.DATEONLY,        field: 'previous_maturity_date' },
  newMaturityDate:       { type: DataTypes.DATEONLY,        field: 'new_maturity_date' },

  previousRemarks:       { type: DataTypes.TEXT,            field: 'previous_remarks' },
  newRemarks:            { type: DataTypes.TEXT,            field: 'new_remarks' },

  blockReason:           { type: DataTypes.TEXT,            field: 'block_reason' },
  policeReportNumber:    { type: DataTypes.STRING,          field: 'police_report_number' },
  policeReportDate:      { type: DataTypes.DATEONLY,        field: 'police_report_date' },

  createdAt:             { type: DataTypes.DATE, defaultValue: DataTypes.NOW, field: 'created_at' },
}, {
  tableName: 'transaction_edit_history',
  timestamps: false,
});

module.exports = TransactionEditHistory;
